docs/reference/indices/diskusage.asciidoc

```diff
@@ -0,0 +1,194 @@
+[[indices-disk-usage]]
+=== Analyze index disk usage API
+++++
+<titleabbrev>Analyze index disk usage</titleabbrev>
+++++
+
+experimental[]
+
+Analyzes the disk usage of each field of an index. This API
+might not support indices created in previous {es} versions.
+The result of a small index can be inaccurate as some parts
+of an index might not be analyzed by the API. It's recommended
+to execute this API with a single index (or the latest backing
+index of a data streams) as the API consumes resources significantly.
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:messages]
+
+[[analyze-index-disk-usage-api-request]]
+==== {api-request-title}
+
+`POST /<index>/_disk_usage`
```

Comment:
Since the API can target data streams. I'd use `<target>` instead of `<index>`.

```suggestion
`POST /<target>/_disk_usage`
```

Reply:
`index` is used many places in the API to mean "index, index alias, or data stream" (`search`, `index`, `create`, `indices.get_mapping`, etc) and `target` is typically used when taking something from one source and applying it elsewhere (see `indices.shrink`, `indices.clone`, `indices.split`) so unsure if `target` would be consistent with other areas of our API spec?

Definitely agree with the change to document that this API can take data streams as a parameter.

Reply:
> `index` is used many places in the API to mean "index, index alias, or data stream" (`search`, `index`, `create`, `indices.get_mapping`, etc)

We use `<target>` in the docs for APIs that can target both data streams and indices:

* [Search API](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-search.html)
* [Index API](https://www.elastic.co/guide/en/elasticsearch/reference/master/docs-index_.html)
* [Get mapping](https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-get-mapping.html)

Using `<index>` as a parameter that can accept data streams is confusing as there are a few APIs that only accept an index, such as the [create index API](https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-create-index.html).

Reply:
Ah my apologies! I read this as a comment on the API spec. Disregard me and carry on!